src/integTest/kotlin/io/snyk/SnykBulkFileListenerTest.kt

```diff
@@ -70,16 +76,31 @@ class SnykBulkFileListenerTest : BasePlatformTestCase() {
 
     @Test
     fun testCurrentIacResults_shouldDropCachedResult_whenIacSupportedFileChanged() {
-        val fakeIacResult = IacResult(null, null)
+        val file = "k8s-deployment.yaml"
+        val filePath = "/src/$file"
+        val iacIssuesForFile = IacIssuesForFile(emptyArray(), file, filePath, "npm")
+        val iacVulnerabilities = arrayOf(iacIssuesForFile)
+        val fakeIacResult = IacResult(iacVulnerabilities, null)
         val toolWindowPanel = project.service<SnykToolWindowPanel>()
         toolWindowPanel.currentIacResult = fakeIacResult
+        val rootIacIssuesTreeNode = toolWindowPanel.getRootIacIssuesTreeNode()
+        rootIacIssuesTreeNode.add(IacFileTreeNode(iacIssuesForFile, project))
 
-        myFixture.configureByText("k8s-deployment.yaml", "some text")
+        myFixture.configureByText(file, "some text")
 
-        assertNull(
-            "cached IacResult should be dropped after IaC supported file changed",
-            toolWindowPanel.currentIacResult
-        )
+        await().atMost(2, TimeUnit.SECONDS)
```

Comment:
JB should have some test util for that. Please check it to avoid adding dependency just for that.

Reply:
They don't have anything besides a Thread.sleep (see https://plugins.jetbrains.com/docs/intellij/testing-faq.html#how-to-wait-for-specified-amount-of-time). This functionality is waiting up to 2s, instead of always 2s, polling every 100ms. The dependency is only for testImplementation, so it does not affect plugin size.

Reply:
Could java.util.concurrent.Future#get serve us here? See example here: ConsoleCommandRunnerTest#testErrorReportedWhenExecutionTimeoutExpire